### PR TITLE
fix healthmgr metrics

### PR DESCRIPTION
**Pull request by huijunw**
after #2821

This PR fix the metrics reporting for healthmgr

after #2853
will refactor this pr after #2853

### CODE REVIEW

File: heron/healthmgr/src/java/com/twitter/heron/healthmgr/HealthManagerMetrics.java [hunk 1]
```diff
@@ -136,44 +168,92 @@ public void run() {
...
+      // The timeout would be the reconnect-interval-seconds
+      sendRequest(request, null, Metrics.MetricPublisherRegisterResponse.newBuilder(),
+          systemConfig.getInstanceReconnectMetricsmgrInterval());
     }
 
     @Override
     public void onResponse(StatusCode status, Object ctx, Message response) {
-      // TODO Auto-generated method stub
+      if (status != StatusCode.OK) {
+        // TODO:- is this a good thing?
```
Comment (nlu90) [inline heron/healthmgr/src/java/com/twitter/heron/healthmgr/HealthManagerMetrics.java]: Is this TODO necessary?

**Comment by huijunwu** [comment]
@nlu90  #2821 is big, this PR is small. so i think we may merge the big then refactor this small.

File: heron/healthmgr/src/java/com/twitter/heron/healthmgr/HealthManagerMetrics.java (hunk 1 above)
Comment (huijunwu) [reply heron/healthmgr/src/java/com/twitter/heron/healthmgr/HealthManagerMetrics.java]: not necessary. will remove it

File: heron/executor/src/python/heron_executor.py [hunk 2]
```diff
@@ -498,7 +498,11 @@ def _get_healthmgr_cmd(self):
                      "--cluster", self.cluster,
                      "--role", self.role,
                      "--environment", self.environment,
-                     "--topology_name", self.topology_name, "--verbose"]
+                     "--topology_name", self.topology_name,
+                     "--metricsmgr_port", self.metrics_manager_port,
+                     "--system_config_file", self.heron_internals_config_file,
+                     "--override_config_file", self.override_config_file,
+                     "--verbose"]
```
[the remark this answers is not in the document]
Comment (huijunwu) [reply heron/executor/src/python/heron_executor.py]: updated

File: heron/healthmgr/src/java/com/twitter/heron/healthmgr/HealthManager.java [hunk 3]
```diff
@@ -211,11 +219,13 @@ public static void main(String[] args) throws Exception {
     LOG.info("Initializing health manager");
     healthManager.initialize();
 
-    LOG.info("Starting Health Manager metirc posting thread");
     HealthManagerMetrics publishingMetricsRunnable = null;
     if (hasOption(cmd, CliArgs.METRICSMGR_PORT)) {
-      publishingMetricsRunnable = new HealthManagerMetrics(
-          Integer.valueOf(getOptionValue(cmd, CliArgs.METRICSMGR_PORT)));
+      LOG.info("Starting Health Manager metirc posting thread");
```
[the remark this answers is not in the document]
Comment (huijunwu) [reply heron/healthmgr/src/java/com/twitter/heron/healthmgr/HealthManager.java]: updated

File: heron/healthmgr/src/java/com/twitter/heron/healthmgr/HealthManager.java [hunk 4]
```diff
@@ -276,10 +283,8 @@ public void initialize() throws ReflectiveOperationException, FileNotFoundExcept
 
     stateMgrAdaptor = createStateMgrAdaptor();
 
-    this.runtime = Config.newBuilder()
-        .put(Key.SCHEDULER_STATE_MANAGER_ADAPTOR, stateMgrAdaptor)
-        .put(Key.TOPOLOGY_NAME, Context.topologyName(config))
-        .build();
+    this.runtime = Config.newBuilder().put(Key.SCHEDULER_STATE_MANAGER_ADAPTOR, stateMgrAdaptor)
+        .put(Key.TOPOLOGY_NAME, Context.topologyName(config)).build();
```
[the remark this answers is not in the document]
Comment (huijunwu) [reply heron/healthmgr/src/java/com/twitter/heron/healthmgr/HealthManager.java]: updated to keep the original format

File: heron/executor/src/python/heron_executor.py [hunk 5]
```diff
@@ -498,7 +498,11 @@ def _get_healthmgr_cmd(self):
                      "--cluster", self.cluster,
                      "--role", self.role,
                      "--environment", self.environment,
-                     "--topology_name", self.topology_name, "--verbose"]
+                     "--topology_name", self.topology_name,
+                     "--metricsmgr_port", self.metrics_manager_port,
+                     "--system_config_file", self.heron_internals_config_file,
```
[the remark this answers is not in the document]
Comment (huijunwu) [reply heron/executor/src/python/heron_executor.py]: updated

File: heron/healthmgr/src/java/com/twitter/heron/healthmgr/HealthManager.java [hunk 6]
```diff
@@ -195,6 +195,14 @@ public static void main(String[] args) throws Exception {
 
     setupLogging(cmd, config);
 
+    LOG.fine(Arrays.toString(cmd.getOptions()));
+
+    // Add the SystemConfig into SingletonRegistry
+    SystemConfig systemConfig = SystemConfig.newBuilder(true)
+        .putAll(getOptionValue(cmd, CliArgs.SYSTEM_CONFIG_FILEPATH), true)
+        .putAll(getOptionValue(cmd, CliArgs.OVERRIDE_CONFIG_FILEPATH), true).build();
+    SingletonRegistry.INSTANCE.registerSingleton(SystemConfig.HERON_SYSTEM_CONFIG, systemConfig);
```
[the remark this answers is not in the document]
Comment (huijunwu) [reply heron/healthmgr/src/java/com/twitter/heron/healthmgr/HealthManager.java]: updated to guice

**Comment by huijunwu** [comment]
Talked with @ashvina yesterday.
Besides reporting the invocation counts, we may also report the results (output of sensor/detector/diagnoser/resolver) in state-cache, which will be added in another PR.

File: heron/healthmgr/src/java/org/apache/heron/healthmgr/HealthManager.java [hunk 7]
```diff
@@ -329,6 +337,8 @@ protected void configure() {
         bind(String.class)
             .annotatedWith(Names.named(CONF_METRICS_SOURCE_TYPE))
             .toInstance(type);
+        bind(HealthManagerMetrics.class)
+            .toInstance(publishingMetricsRunnable);
```
Comment (ashvina) [inline heron/healthmgr/src/java/org/apache/heron/healthmgr/HealthManager.java]: Is naming this instance a "Runnable" necessary? What do you think about naming it just `metricsPublisher`.

Comment (huijunwu) [reply heron/healthmgr/src/java/org/apache/heron/healthmgr/HealthManager.java]: done

**Pull request merged**